### PR TITLE
CAP-51 Fantasy Week

### PR DESCRIPTION
**Pull request by seb8stien1**
## Description

**JIRA Ticket:** [CAP-51](https://mfslcapstone.atlassian.net/browse/CAP-51)

**Screenshots/Examples of Changes:**

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-functional change which does not add or remove functionality)

## Checklist

- [ ] My branch and PR name follow the [naming conventions](https://mfslcapstone.atlassian.net/wiki/spaces/SD/pages/4358145/Development+Naming+Conventions) the team has established.
- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules. (If not, this should be a draft PR)
- [ ] I have [rebased](https://mfslcapstone.atlassian.net/wiki/spaces/SD/pages/4456479/How+To%3A+Rebase+Branch+Onto+Main) my branch onto main before opening this PR, and resolved any merge conflicts.
- [ ] I have ensured that new and existing integrations work with my changes.
- [ ] I have set `@EnableMethodSecurity(securedEnabled)` to true in SecurityConfiguration.java

**Dependencies (if applicable):**

---

[CAP-51]: https://mfslcapstone.atlassian.net/browse/CAP-51?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by linamoufarreh — APPROVED** [review]

File: src/main/java/capstone/mfslbackend/service/FantasyWeekService.java [hunk 1]
```diff
@@ -1,29 +1,223 @@
...
+        return getFantasyWeekByDate(fantasyLeagueId, fantasyTeamId, LocalDate.now());
+    }
+    public FantasyWeek getFantasyWeekByDate(long fantasyLeagueId, long fantasyTeamId, LocalDate date) {
+        FantasyLeague league = fantasyLeagueService.getFantasyLeagueById(fantasyLeagueId);
+        FantasyTeam team = league.getFantasyTeams().stream()
+                .filter(fantasyTeam -> fantasyTeam.getId().equals(fantasyTeamId))
+                .findFirst()
+                .orElseThrow(() -> new Error404("Could not find team with id " + fantasyTeamId + " in league with id " + fantasyLeagueId));
+        return team.getFantasyWeeks().stream()
+                .filter(fantasyWeek -> fantasyWeek.getStartDate().isBefore(LocalDate.now()) && fantasyWeek.getEndDate().isAfter(LocalDate.now()))
```
Comment (abhijeetpraveen) [inline src/main/java/capstone/mfslbackend/service/FantasyWeekService.java:81]: @seb8stien1 fix date

File: src/main/java/capstone/mfslbackend/service/FantasyWeekService.java [hunk 2]
```diff
@@ -1,29 +1,223 @@
...
+        List<Player> bench = players.subList(STARTING_XI, players.size());
+        for (Player player : startingXI) {
+            List<PlayerGameStats> playerStats = stats.stream()
+                    .filter(playerGameStats -> playerGameStats.getPlayer().getPlayerId().equals(player.getPlayerId()))
+                    .toList();
+            if (playerStats.stream()
+                    .map(PlayerGameStats::getMinutes)
+                    .reduce(0, Integer::sum) > 0) {
+                sum += playerStats.stream()
+                        .map(PlayerGameStats::getMinutes)
```
Comment (abhijeetpraveen) [inline src/main/java/capstone/mfslbackend/service/FantasyWeekService.java:164]: @seb8stien1 should be get points

File: src/main/java/capstone/mfslbackend/service/PlayerGameStatsService.java [hunk 3]
```diff
@@ -109,20 +111,19 @@ public List<PlayerGameStats> getPlayerGameStatsByPlayerId(Long playerId) {
         Player player = playerService.getPlayerById(playerId);
         return playerGameStatsRepository.findByPlayer(player);
     }
-    public List<PlayerGameStats> createAllPlayerGameStatsBetweenDates(LocalDate startDate, LocalDate endDate) {
+    public void createAllPlayerGameStatsBetweenDates(LocalDate startDate, LocalDate endDate) {
         List<PlayerGameStats> playerGameStats = new ArrayList<>();
```
Comment (abhijeetpraveen) [inline src/main/java/capstone/mfslbackend/service/PlayerGameStatsService.java:115]: @seb8stien1 delete

File: src/main/java/capstone/mfslbackend/service/PlayerGameStatsService.java [hunk 4]
```diff
@@ -109,20 +111,19 @@ public List<PlayerGameStats> getPlayerGameStatsByPlayerId(Long playerId) {
...
             } catch (Exception e) {
                 e.printStackTrace();
             }
         });
-        return playerGameStats;
     }
 
-    public List<PlayerGameStats> createPlayerGameStats(String fixtureId) {
+    public void createPlayerGameStats(String fixtureId) {
         List<PlayerGameStats> playerGameStats = new ArrayList<>();
```
Comment (abhijeetpraveen) [inline src/main/java/capstone/mfslbackend/service/PlayerGameStatsService.java:127]: Delete

**Review by abhijeetpraveen — COMMENTED** [review]
Requested Changes

**Review by abhijeetpraveen — COMMENTED** [review]
Looks good, let's work on why accepted trades arent working and messing up the lineup. Even accepted trades are going into the rejected state, a problem with the approve team.

**Review by abhijeetpraveen — APPROVED** [review]
LGTM!

**Pull request merged**